.github/workflows/cpp_cron.yml

```diff
@@ -36,170 +36,69 @@ env:
 
 jobs:
 
-  debian:
-    name: AMD64 Debian ${{ matrix.debian }} C++
+  docker:
+    name: ${{ matrix.title }}
     runs-on: ubuntu-latest
     if: ${{ !contains(github.event.pull_request.title, 'WIP') && github.repository == 'apache/arrow' }}
     strategy:
       fail-fast: false
       matrix:
-        debian: [10]
+        name:
+          - debian-10-cpp
+          - fedora-30-cpp
+          - ubuntu-16.04-cpp
+          - ubuntu-18.04-cpp
+          - ubuntu-18.04-cpp-cmake32
+        include:
+          - name: debian-10-cpp
+            image: debian-cpp
+            title: AMD64 Debian 10 C++
+            debian: 10
+          - name: fedora-30-cpp
+            image: fedora-cpp
+            title: AMD64 Fedora 30 C++
+            fedora: 30
+          - name: ubuntu-16.04-cpp
+            image: ubuntu-cpp
+            title: AMD64 Ubuntu 16.04 C++
+            ubuntu: 16.04
+          - name: ubuntu-18.04-cpp
+            image: ubuntu-cpp
+            title: AMD64 Ubuntu 18.04 C++
+            ubuntu: 18.04
+          - name: ubuntu-18.04-cpp-cmake32
+            image: ubuntu-cpp-cmake32
+            title: AMD64 Ubuntu 18.04 C++ CMake 3.2
+            ubuntu: 18.04
     env:
-      DEBIAN: ${{ matrix.debian }}
+      # the defaults here should correspond to the values in .env
```

Comment:
Why are the defaults required, since each matrix entry lists the required value explicitly?

Reply:
The `docker-compose.yml` doesn't contain default values for these environment variables, they are grabbed from the dotenv file. 

If we don't provide a default value for the env variable in the GHA yml then it'll set it to empty. Since we need to explicitly list the docker volumes in the docker-compose.yml:

```
  amd64-ubuntu-14.04-cache:
  amd64-ubuntu-16.04-cache:
  amd64-ubuntu-18.04-cache:
  amd64-ubuntu-20.04-cache:
```

So an empty UBUNTU variable would look for `amd64-ubuntu--cache` which is not defined, so docker-compose bails out.

In my upcoming GHA cache PR I change the type of the docker volumes which may let us to remove the explicit volume definitions, so perhaps we can remove the default env values from the workflow ymls as well.